packages/block-editor/src/components/block-settings-menu/block-settings-dropdown.js

```diff
@@ -174,8 +175,8 @@ export function BlockSettingsDropdown( {
 								: Children.map( ( child ) =>
 										cloneElement( child, { onClose } )
 								  ) }
-							<MenuGroup>
-								{ ! isLocked && (
+							{ canRemove && (
+								<MenuGroup>
```

Comment:
I switched the order here between MenuGroup and isLocked (now canRemove) because when removing was disabled, there would be an empty cell there.
![image](https://user-images.githubusercontent.com/6165348/120818047-27bece80-c54a-11eb-965a-f02071d4f1ce.png)